packages/search/src/content/index.ts

```diff
@@ -131,11 +176,32 @@ function performHubContentSearch(
     headers: {
       authentication: authentication
         ? JSON.stringify(authentication)
-        : undefined
+        : undefined,
     },
     httpMethod: "POST",
-    params: requestParams
+    params: requestParams,
   }).then((response: any) =>
     convertHubResponse(requestParams, response, authentication)
   );
 }
+
+function getSiteCatalogFromOptions(options: IContentSearchOptions) {
+  if (!options || !options.site) return null;
+
+  const ro = getHubRequestOptions(options);
+
```

Comment:
We need to add logic here to accept a wide range of values for `site`:
- `opendata.dc.gov` ideal case
- '3adff78f286e4ecf936bdefdfeca9890' (isGuid(...) - skip domain lookup, AGO & Ent)
- `http://opendata.dc.gov` - drop protocol
- `https://opendata.dc.gov/other/route` - drop protocol and split off trailing paths
- `beta-dcgis` (Enterprise) - ensure no `.`'s

Reply:
@dbouwman thanks for pointing this out.

Reply:
I think this calls for a standalone `fetchSite(idenifier: string): ISiteModel` function where `identifier` is an id (guid), domain (hostname), URL, or enterprise id(, slug?) that does all that. Then this fn calls `fetchSite(options.site)`.